src/Sign.Core/SignatureProviders/ISignatureProvider.cs

```diff
@@ -8,5 +8,10 @@ internal interface ISignatureProvider
     {
         bool CanSign(FileInfo file);
         Task SignAsync(IEnumerable<FileInfo> files, SignOptions options);
+        // Some signature mechanisms (e.g. ClickOnce) require extra files alongside the main file to be signed.
+        // We can't rely on the user specifying everything (and even if we did, we sign all inputs in parallel
+        // so we'd have to add extra synchronisation) so this method instructs an implementation to grab all
+        // dependencies of a file and copy them to the specified directory.
+        void CopySigningDependencies(FileInfo file, DirectoryInfo destination, SignOptions options) { }
```

Comment:
Are there tests that cover usage of this to ensure it's called where it's needed? Also tests for any implementation that ensure the right files are detected and copied to the expected destination?

Reply:
I thought that it was covered by the existing clickonce tests but you're right, it's not since the test just constructs an already-valid directory structure. I'll add some.

Reply:
Pushed a commit with a test for the clickonce-specific behaviour. None of the other signature providers need it (aside from the aggregate that's just passing the call down). The `Signer` class is the only place it's called from and all the signing calls end up there eventually so that should be sufficient I think.